src/util/renderers/parsoidHtmlRestApi.renderer.ts

```diff
@@ -0,0 +1,53 @@
+import domino from 'domino'
+import { Renderer } from './abstractRenderer.js'
+import { getStrippedTitleFromHtml } from '../misc.js'
+import { MWRendererArgs } from './article.renderer.js'
+
+// Represent 'https://{wikimedia-wiki}/api/rest_v1/page/html/'
+export class ParsoidHtmlRestApiRenderer extends Renderer {
+  private data
+  private articleId
+  private articleDetail
+  private articleDetailXId
+
+  constructor(mwRendererArgs: MWRendererArgs) {
+    super()
+    this.data = mwRendererArgs.data
+    this.articleId = mwRendererArgs.articleId
+    this.articleDetail = mwRendererArgs.articleDetail
+    this.articleDetailXId = mwRendererArgs.articleDetailXId
+  }
+
+  public async render(): Promise<any> {
```

Comment:
I don't get it (I probably don't have invested enough time), but here some kind of code in the abstract class should be called IMO?!

Reply:
At the moment there is no code in the render method that can be shared between current renderers.`abstract class Renderer ` has `abstract render(renderOpts: any)` for now, but it can be updated with common functionality when needed.